src/java.base/share/classes/java/security/cert/CertPath.java

```diff
@@ -181,16 +181,9 @@ public boolean equals(Object other) {
         if (this == other)
             return true;
 
-        if (! (other instanceof CertPath))
-            return false;
-
-        CertPath otherCP = (CertPath) other;
-        if (! otherCP.getType().equals(type))
-            return false;
-
-        List<? extends Certificate> thisCertList = this.getCertificates();
-        List<? extends Certificate> otherCertList = otherCP.getCertificates();
-        return(thisCertList.equals(otherCertList));
+        return other instanceof CertPath that
+                && this.type.equals(that.getType())
```

Comment:
I know `type` should never be null but let's change as little as possible by using `that.getType().equals(this.type)`.

Reply:
Ordering swapped around. See cbf2841